Probe/src/probe/rpc.go

```diff
@@ -0,0 +1,136 @@
+/*
+ *  Copyright 2020 Google LLC
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package probe
+
+import (
+	"context"
+	"errors"
+	"google.golang.org/grpc/status"
+	"time"
+
+	"github.com/FirebaseExtended/fcm-external-prober/Controller/src/controller"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+)
+
+const registerRetries int = 10
+const registerTimeout = 10 * time.Second
+
+var client controller.ProbeCommunicatorClient
+var pingConfig *controller.PingConfig
+var hostname string
+
+func initClient() error {
+	tls, err := credentials.NewClientTLSFromFile("cert.pem", "")
+	if err != nil {
+		return err
+	}
+	//TODO(langenbahn): make the internal DNS address of the controller, port, and tls certificate available to the probe
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	conn, err := grpc.DialContext(ctx, ":", grpc.WithTransportCredentials(tls), grpc.WithBlock())
```

Comment:
Does ":" mean this is dialling itself?

Reply:
That's just a placeholder at the moment-- The probe does not yet have access to the information that would allow it to connect to the controller (the controller's hostname, tls certificate, and port), as it has not yet been implemented.